test/uk/gov/hmrc/cgtpropertydisposalsfrontend/connectors/upscan/UpscanConnectorImplSpec.scala

```diff
@@ -31,46 +34,72 @@ import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.generators.FileUploadGen.
 import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.generators.Generators.sample
 import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.generators.IdGen._
 import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.upscan.{UploadReference, UpscanInitiateRequest, UpscanUpload}
-import uk.gov.hmrc.http.{HeaderCarrier, HttpResponse}
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.util.WireMockMethods
+import uk.gov.hmrc.http.test.WireMockSupport
+import uk.gov.hmrc.http.{HeaderCarrier, HttpClient, HttpResponse}
 import uk.gov.hmrc.play.bootstrap.config.ServicesConfig
 
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
 
-class UpscanConnectorImplSpec extends AnyWordSpec with Matchers with MockFactory with HttpSupport {
+class UpscanConnectorImplSpec
+  extends AnyWordSpec
+    with Matchers
+    with MockFactory
+    with GuiceOneAppPerSuite
+    with WireMockSupport
+    with WireMockMethods
+    with EitherValues with HttpSupport {
 
   private val config = Configuration(
     ConfigFactory.parseString(
-      """
+      s"""
         | self {
         |   url = host1.com
         |  },
         |  microservice {
         |    services {
-        |      upscan-initiate {
-        |        protocol = https
-        |        host     = host2
-        |        port     = 123
-        |        user-agent = agent
-        |        max-file-size = 1234
-        |      },
         |      cgt-property-disposals {
-        |        protocol = https
-        |        host     = host3
-        |        port     = 123
+        |        protocol = http
+        |        host     = $wireMockHost
+        |        port     = $wireMockPort
         |      }
         |   }
         |}
         |""".stripMargin
     )
   )
 
-  val connector =
-    new UpscanConnectorImpl(
-      mockHttp,
-      config,
-      new ServicesConfig(config)
+  private val initiateConfig = Configuration(
+    ConfigFactory.parseString(
+      s"""
+         | self {
+         |   url = host1.com
+         |  },
+         |  microservice {
+         |    services {
+         |      upscan-initiate {
+         |        protocol = http
+         |        host     = $wireMockHost
+         |        port     = $wireMockPort
+         |        user-agent = agent
+         |        max-file-size = 1234
+         |      },
+         |     cgt-property-disposals {
+         |        protocol = https
```

Comment:
You could combine these 2 configurations into one. There should be no issue in 2 services being proxied by same wiremock server instance. as the paths we are mocking will be different across these 2 services (cgt-pd & upscan-initiative).